controllers/controllers_suite_test.go

```diff
@@ -92,7 +93,7 @@ func setup() {
 		panic(fmt.Sprintf("unable to setup ClusterCacheTracker: %v", err))
 	}
 
-	controllerOpts := controller.Options{MaxConcurrentReconciles: 10}
+	controllerOpts := controller.Options{MaxConcurrentReconciles: 10, SkipNameValidation: ptr.To(true)}
```

Comment:
Out of curiosity, why do we need this?

Reply:
controller-runtime added a new safeguard and it had to be done with a package-level singleton.

Basically if two controllers are build with the same name the second one will fail.

This was done to ensure that nobody builds & uses two controllers with the same name (to ensure unique metrics & logs).

Unfortunately that is a bit annoying in tests if controllers are build with the same name sequentially (package-level singleton detects that)

Reply:
xref: https://github.com/kubernetes-sigs/controller-runtime/pull/2902